lib/connection_controller.go

```diff
@@ -32,8 +43,12 @@ type ConnectionController struct {
 	// it's aware of at random and provides it to us.
 	AddrMgr *addrmgr.AddrManager
 
-	// When --connectips is set, we don't connect to anything from the addrmgr.
+	// When --connect-ips is set, we don't connect to anything from the addrmgr.
 	connectIps []string
+	// persistentIpToRemoteNodeIdsMap maps persistent IP addresses, like the --connect-ips, to the RemoteNodeIds of the
+	// corresponding RemoteNodes. This is used to ensure that we don't connect to the same persistent IP address twice.
+	// And that we can reconnect to the same persistent IP address if we disconnect from it.
+	persistentIpToRemoteNodeIdsMap map[string]RemoteNodeId
```

Comment:
Consider tracking this in the RemoteNodeManager. If we're going to index and dedupe RemoteNodex by some key, that seems to be the most relevant place to do it

Reply:
Hmm, I thought about it, and it seems like keeping it in the NetworkManager is a better choice, given the `connectIps` map is part of the NetworkManager. It's also more convenient to iterate through this map when it is in the NetworkManager.

Reply:
I changed it to ConcurrentMap by the way.